src/main/protobuf/spawn.proto

```diff
@@ -251,34 +262,57 @@ message ExecLogEntry {
     string target_path = 2;
   }
 
+  message RunfilesTree {
+    // The runfiles tree path.
+    string path = 1;
+    // The entry ID of the set of artifacts in the runfiles tree that are
+    // symlinked at their canonical locations relative to the tree path.
+    // See SpawnLogReconstructor#getRunfilesPaths for how to recover the
+    // tree-relative paths of the artifacts from their exec paths.
+    //
+    // In case of path collisions, later artifacts overwrite earlier ones and
+    // artifacts override custom symlinks.
+    //
+    // The referenced set must not transitively contain any runfile trees.
+    uint32 input_set_id = 2;
+    // The paths relative to the tree path that are symlinked to custom
+    // locations. The target of a symlink is either a file, a directory, an
+    // unresolved symlink (all referenced by ID), or an empty file (represented
+    // by an id of 0).
+    // TODO: If explicitly storing empty files turns out to be too costly,
+    //  consider implementing the logic of the empty file supplier in the
+    //  reconstructor.
+    // TODO: If flattening the symlinks and root symlinks turns out to be too
```

Comment:
I asked on Bazel slack and learned of some plans to maybe encode an entire Python venv into runfiles symlinks. That would certainly void the benefits of this PR. 

What do you think, should we optimize this case now or wait for this to become an issue in practice and then "fix" it by introducing a new field that encodes symlinks as a nested set of entries instead?

Reply:
Does this nested set already exist somewhere in Bazel memory (it doesn't look like it), or would this project also entail rearchitecting the internal representation of runfile trees? The one thing I'm concerned about is prematurely designing for a future that will never arrive, or that will arrive in a different form.